### ISSUE TITLE
"main" field in package.json refers to missing file

### ISSUE THREAD

**Issue by alexanderankin** [farazahmad759/dricup-cli#8]
currently, this package doesn't make anything available to its programmatic interface:

```js
user@localhost:~/test1$ node
Welcome to Node.js v14.12.0.
Type ".help" for more information.
> require("@farazahmad759/dricup-crud-express")
Uncaught:
Error: Cannot find module '/home/user/test1/node_modules/@farazahmad759/dricup-crud-express/src/index.js'. Please verify that the package.json has a valid "main" entry
    at tryPackage (internal/modules/cjs/loader.js:321:19)
    at Function.Module._findPath (internal/modules/cjs/loader.js:534:18)
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:27)
    at Function.Module._load (internal/modules/cjs/loader.js:743:27)
    at Module.require (internal/modules/cjs/loader.js:965:19)
    at require (internal/modules/cjs/helpers.js:88:18) {
  code: 'MODULE_NOT_FOUND',
  path: '/home/user/test1/node_modules/@farazahmad759/dricup-crud-express/package.json',
  requestPath: '@farazahmad759/dricup-crud-express'
}
> 
```

and after uninstalling:

```js
user@localhost:~/test1$ node
Welcome to Node.js v14.12.0.
Type ".help" for more information.
> require("@farazahmad759/dricup-crud-express")
Uncaught Error: Cannot find module '@farazahmad759/dricup-crud-express'
Require stack:
- <repl>
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:893:15)
    at Function.Module._load (internal/modules/cjs/loader.js:743:27)
    at Module.require (internal/modules/cjs/loader.js:965:19)
    at require (internal/modules/cjs/helpers.js:88:18) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ '<repl>' ]
}
```

This is because the `"main"` entry in `package.json`, `src/index.js` does not exist.

It seems like either the `package.json` should not contain the field or it should point to a file which would throw the exception telling the user to user the CLI interface instead.

**Comment by alexanderankin** [comment]
I was actually doing this because i wanted to write some tests and start with `require`ing the module and trying to generate some code generated outputs based on human friendly json schema as in #2, but I also noticed that there was no compilation step - even if there was an index file which exported everything, module support is presently limited enough that you cannot require it from the repl, gives:

```
SyntaxError: Cannot use import statement inside the Node.js REPL, alternatively use dynamic import
```

and taking that advice yields this:

```
> import('./generators/index.js').then(console.log)
Promise { <pending> }
> (node:273306) UnhandledPromiseRejectionWarning: /home/user/dricup-crud-express/src/generators/index.js:1
import dvMigrations from "./migrations/index.js";
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at wrapSafe (internal/modules/cjs/loader.js:992:16)
    at Module._compile (internal/modules/cjs/loader.js:1040:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1097:10)
    at Module.load (internal/modules/cjs/loader.js:941:32)
    at Function.Module._load (internal/modules/cjs/loader.js:782:14)
    at ModuleWrap.<anonymous> (internal/modules/esm/translators.js:139:27)
    at ModuleJob.run (internal/modules/esm/module_job.js:140:23)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async Loader.import (internal/modules/esm/loader.js:165:24)
    at async importModuleDynamicallyWrapper (internal/vm/module.js:443:15)
(node:273306) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 3)

```

so i thought that the right thing to do was to introduce a proper compilation step from tsc to js and expose the compiled js `dist/index.js` instead of `src/index.js` but i am not well-versed enough in the technology to do this. I am not fond of the module system overall.

**Comment by farazahmad759** [comment]
How about requiring like this?
`require("esm")(module /*, options*/)("@farazahmad759/dricup-crud-express").cli(process.argv)`

This works. In fact, bin/dricup-crud-express file uses this same syntax.

EDIT:
There are places for code improvement, that will be done gradually over time while more features are introduced. But for the time being, I think the above method should work. Let me know if this is the case. Thanks.

EDIT 2:
main entry issue in `package.json` has been fixed in `v1.0.7`.

**Comment by alexanderankin** [comment]
nice, didn't notice that; that will do.
